File: internal/content/testdata/bench/go/maximum-subarray-sum.go
func max_subarray_sum(nums []int) int {
	best := nums[0]
	current := nums[0]
	for _, x := range nums[1:] {
		if x > current+x {
			current = x
		} else {
			current = current + x
		}
		if current > best {
			best = current
		}
	}
	return best
}

File: internal/content/testdata/bench/rust/maximum-subarray-sum.rs
fn max_subarray_sum(nums: Vec<i64>) -> i64 {
    let mut best = nums[0];
    let mut current = nums[0];
    for &x in &nums[1..] {
        current = x.max(current + x);
        best = best.max(current);
    }
    best
}
